### ISSUE TITLE
Suricata - please add "bypass" parameter for User defined rules

### ISSUE THREAD

**Issue by Monviech** [opnsense/core#6744]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Is your feature request related to a problem? Please describe.**

Local routed traffic in the opnsense is double inspected by suricata. When NAT is used, the official docs state that the "Interfaces" in "Services: Intrusion Detection: Administration: Settings" should be internal networks. "WAN" shouldn't be selected.
This works well if there is only "WAN" and "LAN".
As soon as there is "WAN", "LAN" and "DMZ", and "LAN", "DMZ" are selected as "Interfaces", the traffic between "LAN" and "DMZ" is inspected when it leaves "LAN" and when it enters "DMZ". This impacts the transmission speed during both inspections.
Creating options for the user to bypass local routed traffic if they only want IPS from/to "$INTERNAL_NET" to/from "$EXTERNAL_NET" would create a huge speed benefit and reduce wasted CPU time.

https://docs.suricata.io/en/suricata-6.0.1/rules/bypass-keyword.html
https://docs.suricata.io/en/suricata-6.0.0/performance/ignoring-traffic.html
https://www.stamus-networks.com/blog/2016/09/28/suricata-bypass-feature

**Describe the solution you like**

I would like an input field in `Services: Intrusion Detection: Administration: User defined` which would add `bypass;` to `/usr/local/etc/suricata/opnsense.rules/OPNsense.rules`.
Rules would look like this:
`pass ip   10.16.1.0/24  any ->  172.16.0.0/24  any (msg:"bypass from 10.16.1.0 to 172.16.0.0"; bypass; sid:1; rev:1;)`

The option would add the `bypass;` parameter to a custom `User defined` rule.

**Describe alternatives you considered**

In `Services: Intrusion Detection: Administration: Settings` there could be an option in the `advanced mode` below `Home networks` which reads `Bypass local traffic from inspection`. The help text would read `Only inspect traffic between "WAN" and "Home networks", bypass inspection between networks defined in "Home networks" completely`.

The setting would read the networks in "Home networks" and create automatic rules between them.

Eg. the networks `192.168.0.0/16, 172.16.0.0/12, a:a:a:a::/64, b:b:b:b::/64` are defined in "Home networks". Following rules would be created automatically in `/usr/local/etc/suricata/opnsense.rules/OPNsense.rules`:

```
pass ip   192.168.0.0/16  any ->  192.168.0.0/16  any (msg:"bypass"; bypass; sid:1; rev:1;)
pass ip   192.168.0.0/16  any ->  172.16.0.0/12  any (msg:"bypass"; bypass; sid:2; rev:1;)
pass ip   172.16.0.0/12  any ->  172.16.0.0/12  any (msg:"bypass"; bypass; sid:3; rev:1;)
pass ip   172.16.0.0/12  any ->  192.168.0.0/16  any (msg:"bypass"; bypass; sid:4; rev:1;)
pass ip  a:a:a:a::/64  any ->  a:a:a:a::/64  any (msg:"bypass"; bypass; sid:5; rev:1;)
pass ip  a:a:a:a::/64  any ->  b:b:b:b::/64  any (msg:"bypass"; bypass; sid:6; rev:1;)
pass ip  b:b:b:b::/64  any ->  b:b:b:b::/64  any (msg:"bypass"; bypass; sid:7; rev:1;)
pass ip  b:b:b:b::/64  any ->  a:a:a:a::/64  any (msg:"bypass"; bypass; sid:8; rev:1;)
```

**Additional context**

I have created a forum post too with additional context:
https://forum.opnsense.org/index.php?topic=35340.0

Thank you for reading this and considering if its a viable option to add.

**Comment by Monviech** [comment]
I'm kinda new to github so I don't know the proper way to do things, but I tried to implement the feature myself in order to get a better understanding.

https://github.com/opnsense/core/compare/master...Monviech:core:master

![grafik](https://github.com/opnsense/core/assets/79600909/e2d11df8-44a5-491b-a907-bc2f6b279d20)

![grafik](https://github.com/opnsense/core/assets/79600909/4fd20afb-f5f9-4fac-95f9-d0d8be304c46)

**Comment by Monviech** [comment]
Closed with #6752

**Comment by Monviech** [comment]
Closed with #6752
